go/libkb/passphrase_login.go

```diff
@@ -0,0 +1,200 @@
+			return nil
+		}
+		if _, badpw := err.(PassphraseError); !badpw {
+			return err
+		}
+		retryMsg = err.Error()
+	}
+	return err
+}
+
+type loginReply struct {
+	Status    AppStatus    `json:"status"`
+	Session   string       `json:"session"`
+	CsrfToken string       `json:"csrf_token"`
+	UID       keybase1.UID `json:"uid"`
+	Me        struct {
+		Basics struct {
+			Username             string               `json:"username"`
+			PassphraseGeneration PassphraseGeneration `json:"passphrase_generation"`
+		} `json:"basics"`
+	} `json:"me"`
+}
+
+func (l *loginReply) GetAppStatus() *AppStatus {
+	return &l.Status
+}
+
+func pplPost(m MetaContext, eOu string, lp PDPKALoginPackage) (*loginAPIResult, error) {
+
+	arg := APIArg{
+		Endpoint:    "login",
+		SessionType: APISessionTypeNONE,
+		Args: HTTPArgs{
+			"email_or_username": S{eOu},
+		},
+		NetContext:     m.Ctx(),
+		AppStatusCodes: []int{SCOk, SCBadLoginPassword, SCBadLoginUserNotFound},
+	}
+	lp.PopulateArgs(&arg.Args)
+	var res loginReply
+	err := m.G().API.PostDecode(arg, &res)
+	if err != nil {
+		return nil, err
+	}
+	if res.Status.Code == SCBadLoginPassword {
+		err = PassphraseError{"server rejected login attempt"}
+		return nil, err
+	}
+	if res.Status.Code == SCBadLoginUserNotFound {
+		return nil, NotFoundError{}
+	}
+	return &loginAPIResult{
+		sessionID: res.Session,
+		csrfToken: res.CsrfToken,
+		uid:       res.UID,
+		username:  res.Me.Basics.Username,
+		ppGen:     res.Me.Basics.PassphraseGeneration,
+	}, nil
+}
+
+func PassphraseLoginPrompt(m MetaContext, usernameOrEmail string, maxAttempts int) (err error) {
+
+	defer m.CTrace("PassphraseLoginPrompt", func() error { return err })()
+
+	var loginSession *LoginSession
+
+	if err = pplPromptCheckPreconditions(m, usernameOrEmail); err != nil {
+		return err
+	}
+	if usernameOrEmail, err = pplGetEmailOrUsername(m, usernameOrEmail); err != nil {
+		return err
+	}
+	if loginSession, err = pplGetLoginSession(m, usernameOrEmail); err != nil {
+		return err
+	}
+	if err = pplPromptLoop(m, usernameOrEmail, maxAttempts, loginSession); err != nil {
+		return err
+	}
+	return nil
+}
+
+func StoreSecretAfterLogin(m MetaContext, n NormalizedUsername, uid keybase1.UID, deviceID keybase1.DeviceID) (err error) {
+	defer m.CTrace("StoreSecretAfterLogin", func() error { return err })()
+	lksec := NewLKSecWithDeviceID(m.LoginContext().PassphraseStreamCache().PassphraseStream(), uid, deviceID, m.G())
+	return StoreSecretAfterLoginWithLKS(m, n, lksec)
+}
+
+func StoreSecretAfterLoginWithLKS(m MetaContext, n NormalizedUsername, lks *LKSec) (err error) {
+
+	defer m.CTrace("StoreSecretAfterLoginWithLKS", func() error { return err })()
+
+	secretStore := NewSecretStore(m.G(), n)
+	if secretStore == nil {
+		m.CDebugf("not storing secret; no secret store available")
+		return nil
+	}
+
+	secret, err := lks.GetSecret(m)
+	if err != nil {
+		return err
+	}
+
+	// Ignore any errors storing the secret.
```

Comment:
Seems like you changed this behavior?

Reply:
the comment is buggy.  The error is ignored where it's called above.

Reply:
ah, you're right, in signup i ignore the error. here i don't.  i'll change it back.

Reply:
thanks for finding this